src/domain/tradingview/helper.ts

```diff
@@ -0,0 +1,120 @@
+import { getServerUrl } from "config/backend";
+import { getTokenBySymbol, getWrappedToken } from "config/tokens";
+import {
+  getChainlinkChartPricesFromGraph,
+  getChartPricesFromStats,
+  getLimitChartPricesFromStats,
+  getStablePriceData,
+  timezoneOffset,
+} from "domain/prices";
+import { CHART_PERIODS, USD_DECIMALS } from "lib/legacy";
+import { formatAmount } from "lib/numbers";
+
+export const supportedResolutions = { 5: "5m", 15: "15m", 60: "1h", 240: "4h", "1D": "1d" };
+const LAST_BAR_REFRESH_INTERVAL = 15000; // 15 seconds
+
+export function getPeriodFromResolutions(value, object = supportedResolutions) {
```

Comment:
I think it's better to split the code here:
1. Put the requests in requests.ts
2. Simple utils which just format something could be be placed in utils.ts

It is also worth  add types to all functions here and in the useTVDatafeed hook

Reply:
Done!